docs/topics/asyncio.rst

```diff
@@ -98,6 +98,52 @@ Futures. Scrapy provides two helpers for this:
          into your own code.
 
 
+Async additional requests
+=========================
+
+The spider below shows a single use-case of scraping a page and gathering a price from a separate URL::
+
+
+    class SingleRequestSpider(scrapy.Spider):
+        name = "single"
+        start_urls = ["https://example.org/product"]
+
+        async def parse(self, response, **kwargs):
+            additional_request = scrapy.Request('https://example.org/price')
+            deferred = self.crawler.engine.download(additional_request)
+            additional_response = await maybe_deferred_to_future(deferred)
+            yield {
+                'h1': response.css('h1').get(),
+                'price': additional_response.css('#price').get(),
+            }
+
+
+The spider gathering batch requests::
+
+    class BatchRequestsSpider(scrapy.Spider):
+        name = "batch"
+        start_urls = ["https://example.com/product"]
+
+        async def parse(self, response, **kwargs):
+            additional_requests = [
+                scrapy.Request("https://example.com/price"),
+                scrapy.Request("https://example.com/color"),
+            ]
+            coroutines = []
+            for r in additional_requests:
+                deffered = self.crawler.engine.download(r)
+                coroutines.append(maybe_deferred_to_future(deffered))
+
+            responses = await asyncio.gather(
+                *coroutines, return_exceptions=True
+            )
+            yield {
+                'h1': response.css('h1::text').get(),
+                'price': responses[0].css('.price::text').get(),
+                'color': responses[1].css('.color::text').get(),
+            }
```

Comment:
I don’t think that will work with the non-asyncio Twisted reactor. Could you see of you can get a working alternative with DeferredList, in line with the following (untested) code?

```suggestion
    from twisted.internet.defer import DeferredList

    class BatchRequestsSpider(scrapy.Spider):
        name = "batch"
        start_urls = ["https://example.com/product"]

        async def parse(self, response, **kwargs):
            additional_requests = [
                scrapy.Request("https://example.com/price"),
                scrapy.Request("https://example.com/color"),
            ]
            deferreds = []
            for r in additional_requests:
                deferred = self.crawler.engine.download(r)
                deferreds.append(deferred)
            responses = await maybe_deferred_to_future(DeferredList(deferreds))
            yield {
                'h1': response.css('h1::text').get(),
                'price': responses[0].css('.price::text').get(),
                'color': responses[1].css('.color::text').get(),
            }
```

Reply:
Fixed. Added some fixes because `response` converted to a list of tuples.

Reply:
Moved imports from twisted to scrapy.

Reply:
> I don’t think that will work with the non-asyncio Twisted reactor. 

That's why it's in asyncio.rst, not in coroutines.rst :P

Reply:
@wRAR @Gallaecio 
As a working sample, we can add `asyncio` reactor in the example:

```
class MultipleRequestsSpider(scrapy.Spider):
    ...
    @classmethod
    def update_settings(cls, settings):
        settings["TWISTED_REACTOR"] = "twisted.internet.asyncioreactor.AsyncioSelectorReactor"

    async def parse(self, response, **kwargs):
        ...
        responses = await asyncio.gather(
            *coroutines, return_exceptions=True
        ...
```

Reply:
Assuming the point is to show how to use inline requests, I think moving the examples to coroutines.rst is the way to go, instead of keeping them asyncio-specific.